### PR TITLE
Fixes issue #40: extraneous zero padding on serialized exponent

### PR DESCRIPTION
**Pull request by csstaub**
@dgalling @r0ro

### CODE REVIEW

File: encoding.go [hunk 1]
```diff
@@ -175,3 +184,15 @@ func (b byteBuffer) bigInt() *big.Int {
 func (b byteBuffer) toInt() int {
 	return int(b.bigInt().Int64())
 }
+
+func trimZeros(data []byte) []byte {
```
Comment (dgalling) [inline encoding.go]: Could you use `bytes.TrimLeft` instead?

Comment (csstaub) [inline encoding.go]: Derp, yes

File: encoding.go [hunk 2]
```diff
@@ -131,6 +132,14 @@ func newBuffer(data []byte) *byteBuffer {
 	}
 }
 
+func newBufferFromInt(num uint64) *byteBuffer {
+	data := make([]byte, 8)
+	binary.BigEndian.PutUint64(data, num)
+	return &byteBuffer{
```
Comment (dgalling) [inline encoding.go]: Minor: I prefer we `return newBuffer(bytes.TrimLeft(data, "\x00"))`, but it's up to you.

Comment (csstaub) [inline encoding.go]: Yep, I like that better too. Will update and merge after lunch

**Pull request merged**